### DISCUSSION TITLE
In android java the audio is not playing from "https://dictionaryapi.dev/"

### DISCUSSION THREAD

**Discussion by momin-sana** [community/community#50676] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

I have already posted the same issue at ### https://github.com/momin-sana/FindMeaning/issues/1
Please help me to solve this issue.

I am creating a dictionary app in **Android java using "https://dictionaryapi.dev/"**. I am fetching data by _Retrofit_, and using _list Adapters_ to display my data. I am facing a problem in playing audio which is being fetched from mentioned API. Please help me to solve this problem, if the issue is with API so how can I solve it to play my audio? If I use an alternative link for audio how to link it to my other code?

_At Debugging_
![226828657-67091508-e64f-45aa-8008-e5db3b0c51a5](https://user-images.githubusercontent.com/92250920/226831879-441f81d8-9882-4833-936d-4d2f2fede958.png)

_At Run: On clicking at the audio icon this error appears on RUN: "E/MediaPlayerNative: error (1, -2147483648) E/MediaPlayer: Error (1,-2147483648)"_
![226828774-8cf85950-0103-4f48-b339-97460a65b321](https://user-images.githubusercontent.com/92250920/226831936-839460e5-1ed6-4970-8c9d-ce5f54e34eff.png)

_The following appears in logcat_
- E/AUDIO_ERROR: MediaPlayer exception: null
- E/MediaPlayerNative: error (1, -2147483648)

My relative code is as followed, where I tried adding different audio sources to check are other APIs are working or not. I came to conclude that YES! all files are working, and my audio is playing except "[https://api.dictionaryapi.dev/media/pronunciations/en/{word}-us.mp3](https://api.dictionaryapi.dev/media/pronunciations/en/%7Bword%7D-us.mp3)".

```
public class PhoneticAdapter extends RecyclerView.Adapter<PhoneticVH> {
    private final Context context;
    private List<Phonetic> phoneticList;
    private MediaPlayer player;

public PhoneticAdapter(Context context, List<Phonetic> phoneticList) {
        this.context = context;
        this.phoneticList = phoneticList;
    }


    @NonNull
    @Override
    public PhoneticVH onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new PhoneticVH(LayoutInflater.from(context).inflate(R.layout.phonetic_list_items, parent, false));
    }

    @Override
    public void onBindViewHolder(@NonNull PhoneticVH holder, @SuppressLint("RecyclerView") int position) {
        holder.phonetic_text.setText(phoneticList.get(position).getText());
        holder.phonetic_audio.setOnClickListener(view -> {
//            String audioUrl = "https://" + phoneticList.get(position).getAudio();
            player = new MediaPlayer();
            try{
                AudioAttributes audioAttributes = new AudioAttributes.Builder()
                        .setUsage(AudioAttributes.USAGE_MEDIA)
                        .setContentType(AudioAttributes.CONTENT_TYPE_MUSIC)
                        .build();

                player.setAudioAttributes(audioAttributes);
//                player.setDataSource("https://www.soundhelix.com/examples/mp3/SoundHelix-Song-1.mp3");
//                player.setDataSource("https://www.learningcontainer.com/wp-content/uploads/2020/02/Kalimba.mp3");
//                player.setDataSource("https://www.learningcontainer.com/wp-content/uploads/2020/02/Kalimba-online-audio-converter.com_-1.wav");
//                player.setDataSource("https://www.learningcontainer.com/wp-content/uploads/2020/02/Sample-FLAC-File.flac");
//                player.setDataSource("https://www.learningcontainer.com/wp-content/uploads/2020/02/Sample-OGG-File.ogg");
              
              player.setDataSource("https://" + phoneticList.get(position).getAudio());
                player.setOnPreparedListener(mp -> {
                    mp.start();
                });
                player.setOnErrorListener((mediaPlayer, i, i1) -> {
                    Log.e("AUDIO_ERROR", "MediaPlayer error: " + i);
                    Toast.makeText(context, "Error: " + i, Toast.LENGTH_SHORT).show();
                    return false;
                });
                player.prepareAsync();
                player.prepare();
                player.start();
            }catch (Exception e){
                Log.e("AUDIO_ERROR", "MediaPlayer exception: " + e.getMessage());
                Toast.makeText(context, R.string.audio_fail, Toast.LENGTH_SHORT).show();
            }
        });
    }

    @Override
    public int getItemCount() {
        if (phoneticList == null) {
            return 0;
        }
        return phoneticList.size();
    }

    @SuppressLint("NotifyDataSetChanged")
    public void setData(List<Phonetic> phoneticList) {
        this.phoneticList = phoneticList;
        notifyDataSetChanged();
    }
}

```

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @momin-sana ! 

We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Issues category is solely related to conversations around the GitHub products Issues and Projects. I've gone ahead and moved it for you. Good luck!
